models/netsuite2/netsuite2__balance_sheet.sql

```diff
@@ -106,10 +114,17 @@ balance_sheet as (
         else accounts.account_id
             end as account_id,
         case
-        when not accounts.is_balancesheet then null
+        when not accounts.is_balancesheet then (select accounts.account_number from accounts where lower(accounts.special_account_type_id) = 'retearnings' limit 1)
```

Comment:
Via @jmongerlyra [in the issue](https://github.com/fivetran/dbt_netsuite/issues/135):

"The account number for income statement and CTA accounts in the balance sheet model is currently set to NULL. This is not how it is presented on the native Balance Sheet report in NetSuite.

Income statement accounts should use the account number of the system-generated retained earnings account. CTA should use the account number of the system-generated CTA account."

This logic ensures that the account number pulls the retained earnings or CTA account number rather than generating null values. I added some details in the CHANGELOG to call this out.

Reply:
This makes sense to me. Thanks for sharing this and calling it out in the CHANGELOG.